lib/state_machine/src/ChargerStateMachine.cpp

```diff
@@ -0,0 +1,111 @@
+#include "ChargerStateMachine.h"
+
+
+//logic for changing states - still need to account for dial_position
+ChargerState_e ChargerStateMachine::evaluate_charger_state_machine(ChargerState_e _current_state, bool balancing_enabled, int dial_position, bool start_button_pressed) 
+{
+    switch (_current_state) //takes in the _current_state variables and matches it to each case
+    {
+        case ChargerState_e::INITIAL:
+        {
+            if (start_button_pressed) {
+                set_state(ChargerState_e::CHARGING_NO_BALANCING);
+            }
+
+            if (balancing_enabled) {
+                set_state(ChargerState_e::BALANCING);
+            } 
+
+            break;
+
+        }
+
+        case ChargerState_e::BALANCING:
+        {
+            if (balancing_enabled) {
+                set_state(ChargerState_e::BALANCING);
```

Comment:
this will end up calling the exit logic and entry logic on every tick, which is unnecessary. I'd just have a
```c++
if (!balancing_enabled) {
    set_state(ChargerState_e::CHARGING_NO_BALANCING)
}
break;
```